### ISSUE TITLE
Prometheus_scrape gives could not process metrics payload error

### ISSUE THREAD

**Issue by krishnashed** [fluent/fluent-bit#6322]
## Bug Report

**Describe the bug**
Wasn't able to run prometheus_scrape ( followed the steps and same config as in example at https://docs.fluentbit.io/manual/pipeline/inputs/prometheus-scrape-metrics )

**To Reproduce**
I have my prometheus running on port 9090
My fluent-bit.conf is 

[INPUT]
    name prometheus_scrape
    host 0.0.0.0
    port 9090
    tag prometheus
    metrics_path /metrics?format=prometheus
    scrape_interval 5s
[OUTPUT]
    name file
    match *
    path path

Prometheus metrics :
![image](https://user-images.githubusercontent.com/60742358/199213076-62497082-333a-4d76-9826-1a8b60a21049.png)

On running the command : fluent-bit -c fluent-bit.conf
It gives error as :
[2022/10/31 18:11:28] [error] [output:file:file.0] could not process metrics payload

**Expected behavior**
Expecting the same example output as given in Documentation.

**Your Environment**

* Version used: fluent-bit v2.0.3
* Environment name and version (e.g. Kubernetes? What version?): minikube v1.27.1 on Arch 22.0.0, k8s v1.25.2
* Operating System and version: Manjaro v22.0.0

**Comment by agup006** [comment]
Thanks for submitting, can you copy the Prometheus text output for us?

**Comment by krishnashed** [comment]
[prometheus_metrics.txt](https://github.com/fluent/fluent-bit/files/9922177/prometheus_metrics.txt)

Here it is

**Comment by nokute78** [comment]
I tested using `out_stdout` and below prometheus.
https://github.com/prometheus/prometheus/releases/tag/v2.39.1

Fluent-bit reported error.
```
[2022/11/03 09:28:07] [error] [output:stdout:stdout.0] could not process metrics payload 
```

It reached below limit.
https://github.com/fluent/fluent-bit/blob/v2.0.3/lib/cmetrics/src/cmt_decode_msgpack.c#L446

**Comment by nokute78** [comment]
I opened an issue on `cmetrics` repo.
https://github.com/fluent/cmetrics/issues/167

**Comment by nokute78** [comment]
https://github.com/fluent/cmetrics/issues/167 is fixed on cmetrics v0.5.7.
Fluent-bit v2.0.5 contains it and the issue should be fixed.

```
$ ../../bin/fluent-bit -c a.conf 
Fluent Bit v2.0.6
* Copyright (C) 2015-2022 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/11/12 09:34:11] [ info] [fluent bit] version=2.0.6, commit=2049e320eb, pid=9279
[2022/11/12 09:34:11] [ info] [storage] ver=1.2.0, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2022/11/12 09:34:11] [ info] [cmetrics] version=0.5.7
[2022/11/12 09:34:11] [ info] [ctraces ] version=0.2.5
[2022/11/12 09:34:11] [ info] [input:prometheus_scrape:prometheus_scrape.0] initializing
[2022/11/12 09:34:11] [ info] [input:prometheus_scrape:prometheus_scrape.0] storage_strategy='memory' (memory only)
[2022/11/12 09:34:11] [ info] [sp] stream processor started
[2022/11/12 09:34:11] [ info] [output:stdout:stdout.0] worker #0 started
2022-11-12T00:34:16.312925650Z go_memstats_alloc_bytes_total = 85754216
2022-11-12T00:34:16.312925650Z go_memstats_frees_total = 81015
2022-11-12T00:34:16.312925650Z go_memstats_lookups_total = 0
2022-11-12T00:34:16.312925650Z go_memstats_mallocs_total = 177507
2022-11-12T00:34:16.312925650Z net_conntrack_dialer_conn_attempted_total{dialer_name="alertmanager"} = 0
2022-11-12T00:34:16.312925650Z net_conntrack_dialer_conn_attempted_total{dialer_name="default"} = 0
2022-11-12T00:34:16.312925650Z net_conntrack_dialer_conn_attempted_total{dialer_name="prometheus"} = 1
2022-11-12T00:34:16.312925650Z net_conntrack_dialer_conn_closed_total{dialer_name="alertmanager"} = 0
2022-11-12T00:34:16.312925650Z net_conntrack_dialer_conn_closed_total{dialer_name="default"} = 0
2022-11-12T00:34:16.312925650Z net_conntrack_dialer_conn_closed_total{dialer_name="prometheus"} = 0
2022-11-12T00:34:16.312925650Z net_conntrack_dialer_conn_established_total{dialer_name="alertmanager"} = 0
2022-11-12T00:34:16.312925650Z net_conntrack_dialer_conn_established_total{dialer_name="default"} = 0
2022-11-12T00:34:16.312925650Z net_conntrack_dialer_conn_established_total{dialer_name="prometheus"} = 1
2022-11-12T00:34:16.312925650Z net_conntrack_dialer_conn_failed_total{dialer_name="alertmanager",reason="refused"} = 0
2022-11-12T00:34:16.312925650Z net_conntrack_dialer_conn_failed_total{dialer_name="alertmanager",reason="resolution"} = 0
```
